islandora_solr_config/includes/atom_results.inc

```diff
@@ -0,0 +1,248 @@
+      foreach ($doc['solr_doc'] as $key => $value) {
+        if (is_array($value)) {
+          // Turn array into comma separated string and trim.
+          $doc[$key] = trim(implode(', ', $value));
+        }
+        else {
+          // Give it a trim.
+          $doc[$key] = trim($value);
+        }
+      }
+      // Get the variables for the <item> element.
+      $item = $this->rssItem($doc);
+
+      // Hook alter to change the rssItem before formatting.
+      drupal_alter('islandora_solr_search_rss_item', $item, $doc);
+
+      // Render rss item.
+      $rendered_item = format_rss_item($item['title'], $item['link'], $item['description'], $item['items']);
+      $open_tag = "<item>";
+      $close_tag = "</item>";
+      $rendered_item = str_replace($open_tag, "<entry>", $rendered_item);
+      $rendered_item = str_replace($close_tag, "</entry>", $rendered_item);
+
+      // ... allow it to be altered...
+      drupal_alter('islandora_solr_config_rss_item_post_render', $rendered_item, $doc);
+
+      // ... and add to items string.
+      $items .= "$rendered_item\n";
+    }
+
+    // Query search terms:
+    $query = $islandora_solr_query->solrQuery;
+
+    // Get the variables for the <channel> element.
+    $channel = $this->rssChannel($query);
+    drupal_alter('islandora_solr_config_rss_root_element_attributes', $channel, $items);
+
+    // Give the results clean variable names.
+    $title = $channel['title'];
+    $url = $channel['url'];
+    $feed_url = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]";
+    if (strpos($feed_url, '&page') == FALSE) {
+      $page_number = '1';
+    }
+    else {
+      $arr = explode('&page=', $feed_url);
+      $page_number = $arr[1];
+    }
+
+    $link_self = $feed_url;
+    if (isset($arr)) {
+      $link_first = $arr[0];
+    }
+    else {
+      $link_first = $feed_url;
+    }
+    $page_next = $page_number + 1;
+    if ($page_number > 1) {
+      $page_prev = $page_number - 1;
+    }
+    $link_next = $link_first . '&page=' . $page_next;
+    if (isset($page_prev)) {
+      $link_prev = $link_first . '&page=' . $page_prev;
+    }
+    $description = $channel['description'];
+    $langcode = $channel['langcode'];
+    $args = $channel['args'];
+
+    $output = "<?xml version=\"1.0\" encoding=\"utf-8\"?>\n";
+    $output .= '<feed xmlns="http://www.w3.org/2005/Atom">' . "\n";
+    $output .= ' <link rel="self" href="' . $link_self . '"/>' . "\n";
+    $output .= ' <link rel="first" href="' . $link_first . '"/>' . "\n";
+    if (isset($link_prev)) {
+      $output .= ' <link rel="previous" href="' . $link_prev . '"/>' . "\n";
+    }
+    $output .= ' <link rel="next" href="' . $link_next . '"/>';
+    $strip_channel = format_rss_channel($title, $url, $description, $items, $langcode, $args);
+    $tags = array(
+      "<channel>",
+      "</channel>",
+    );
+    $output .= str_replace($tags, "", $strip_channel);
```

Comment:
Same comment as above. I think we should probably create a class member `format_atom_channel` and use it instead of calling `format_rss_channel` and then replacing the elements.